src/components/configure/header/Header.tsx

```diff
@@ -97,6 +110,159 @@ export default class Header extends React.Component<HeaderProps, HeaderState> {
   }
 
+  handleAuthMenuIconClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    this.setState({
+      authMenuAnchorEl: event.currentTarget,
+    });
+  };
+
+  handleAuthMenuClose = () => {
+    this.setState({
+      authMenuAnchorEl: null,
+    });
+  };
+
+  async handleLogoutMenuClick() {
+    this.props.logout!();
+    this.setState({
+      authMenuAnchorEl: null,
+    });
+  }
+
+  async handleLoginMenuClick() {
+    this.setState({ authMenuAnchorEl: null, openAuthProviderDialog: true });
+  }
+
+  async handleLinkGoogleAccountMenuClick() {
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGoogleAccount!();
+  }
+
+  async handleLinkGitHubAccountMenuClick() {
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGitHubAccount!();
+  }
+
+  renderLinkGoogleAccountMenu() {
+    const user = this.props.auth!.getCurrentAuthenticatedUser();
+    if (user && !getGoogleProviderData(user).exists) {
+      return (
+        <MenuItem
+          key="1"
+          button={true}
+          onClick={() => this.handleLinkGoogleAccountMenuClick()}
+        >
+          Link Google Account
+        </MenuItem>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  renderLinkGitHubAccountMenu() {
+    const user = this.props.auth!.getCurrentAuthenticatedUser();
+    if (user && !getGitHubProviderData(user).exists) {
+      return (
+        <MenuItem
+          key="2"
+          button={true}
+          onClick={() => this.handleLinkGitHubAccountMenuClick()}
+        >
+          Link GitHub Account
+        </MenuItem>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  renderAvatarIcon() {
+    const { authMenuAnchorEl } = this.state;
+    if (this.props.signedIn) {
+      const user = this.props.auth!.getCurrentAuthenticatedUser();
+      const profileImageUrl = user.photoURL || '';
+      const profileDisplayName = user.displayName || '';
+      let avatar: React.ReactNode;
+      if (profileImageUrl) {
+        avatar = (
+          <Avatar
+            alt={profileDisplayName}
+            src={profileImageUrl}
+            className="header-avatar"
+          />
+        );
+      } else {
+        avatar = (
+          <Avatar className="header-avatar">
+            <Person />
+          </Avatar>
+        );
+      }
+      return (
+        <React.Fragment>
+          <IconButton
+            aria-owns={
+              authMenuAnchorEl ? 'configure-header-auth-menu' : undefined
+            }
+            onClick={this.handleAuthMenuIconClick}
+          >
+            {avatar}
+          </IconButton>
+          <Menu
+            id="configure-header-auth-menu"
+            anchorEl={authMenuAnchorEl}
+            open={Boolean(authMenuAnchorEl)}
+            onClose={this.handleAuthMenuClose}
+          >
+            {this.renderLinkGoogleAccountMenu()}
+            {this.renderLinkGitHubAccountMenu()}
+            <MenuItem
+              key="3"
```

Comment:
It's not problem in this case, for your reference, using the index as a key might be unti-pattern in React. https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
@adamrocker Thank you for the comment! Yes, each value specified by me was such index value. I changed the values at dc1b1fa.